toolbox/arcgee/data.py

```diff
@@ -1054,6 +1054,27 @@ def init_and_set_tags(project=None, workload_tag=None):
     return
 
 
+# Get ROI from the object extent.
+# Use string-based type hints to avoid type checking errors in ArcGIS Pro 3.2.
+def get_roi_from_object(obj: "ee.Image | ee.FeatureCollection") -> "ee.Geometry.BBox":
+    """Get the ROI from the object extent.
+
+    Args:
+        obj (ee.Image or ee.FeatureCollection): Earth Engine object
+
+    Returns:
+        ee.Geometry.BBox: ROI from the object extent
+    """
+    arcpy.AddMessage("Trying to get ROI from the object extent ...")
+    # Get extent of the object.
+    centroid_coords, bounds_coords = get_object_centroid(obj, 1)
+    x_min, y_min, x_max, y_max = convert_coords_to_bbox(bounds_coords)
+    arcpy.AddMessage([x_min, y_min, x_max, y_max])
+    roi = ee.Geometry.BBox(x_min, y_min, x_max, y_max)
+
+    return roi
```

Comment:
No need for the local `roi` variable.

`return ee.Geometry.BBox(x_min, y_min, x_max, y_max)`